app/src/components/market/common/market_scale/index.tsx

```diff
@@ -378,37 +381,66 @@ export const MarketScale: React.FC<Props> = (props: Props) => {
   const [profitLoss, setProfitLoss] = useState(0)
   const [shortPayout, setShortPayout] = useState(0)
   const [longPayout, setLongPayout] = useState(0)
-  const [totalShortPrice, setTotalShortPrice] = useState<number>(0)
-  const [totalLongPrice, setTotalLongPrice] = useState<number>(0)
+  const [totalShortPrice, setTotalShortPrice] = useState<BigNumber>(new BigNumber(0))
+  const [totalLongPrice, setTotalLongPrice] = useState<BigNumber>(new BigNumber(0))
+  const [shortProfitAmount, setShortProfitAmount] = useState(0)
+  const [longProfitAmount, setLongProfitAmount] = useState(0)
+  const [longProfitPercentage, setLongProfitPercentage] = useState(0)
+  const [shortProfitPercentage, setShortProfitPercentage] = useState(0)
 
   useEffect(() => {
+    let totalShortTradesCost = new BigNumber(0)
+    let totalLongTradesCost = new BigNumber(0)
     if (trades && trades.length && collateral) {
       const shortTrades = trades.filter(trade => trade.outcomeIndex === '0')
-      setTotalShortPrice(
-        shortTrades.length
-          ? shortTrades
-              .map(trade =>
-                trade.type === 'Buy'
-                  ? Number(formatBigNumber(trade.collateralAmount, collateral.decimals, collateral.decimals))
-                  : Number(formatBigNumber(trade.collateralAmount, collateral.decimals, collateral.decimals)) * -1,
-              )
-              .reduce((a, b) => a + b)
-          : 0,
-      )
+      totalShortTradesCost = shortTrades.length
+        ? shortTrades
+            .map(trade => (trade.type === 'Buy' ? trade.collateralAmount : Zero.sub(trade.collateralAmount)))
+            .reduce((a, b) => a.add(b))
+        : new BigNumber(0)
+
       const longTrades = trades.filter(trade => trade.outcomeIndex === '1')
-      setTotalLongPrice(
-        longTrades.length
-          ? longTrades
-              .map(trade =>
-                trade.type === 'Buy'
-                  ? Number(formatBigNumber(trade.collateralAmount, collateral.decimals, collateral.decimals))
-                  : Number(formatBigNumber(trade.collateralAmount, collateral.decimals, collateral.decimals)) * -1,
-              )
-              .reduce((a, b) => a + b)
-          : 0,
+      totalLongTradesCost = longTrades.length
+        ? longTrades
+            .map(trade => (trade.type === 'Buy' ? trade.collateralAmount : Zero.sub(trade.collateralAmount)))
```

Comment:
Better to have a type TradeType {Buy: 'buy', Sell: 'sell'} and compare  TradeType.Buy / TradeType.Sell